test_rclcpp/CMakeLists.txt

```diff
@@ -69,6 +69,26 @@ if(AMENT_ENABLE_TESTING)
         "${middleware_impl}"
         "rclcpp")
     endif()
+
+    # test_timeout_subscriber
+    ament_add_gtest(
+      gtest_timeout_subscriber__${middleware_impl}
+      "test/test_timeout_subscriber.cpp"
+      TIMEOUT 30
+    )
```

Comment:
The source file `test_timeout_subscriber.cpp` has a main function and you also link against the default `main` of gtest. This will likely not work.